cmd/stratus/warmup_cmd.go

```diff
@@ -41,11 +42,28 @@ func buildWarmupCmd() *cobra.Command {
 }
 
 func doWarmupCmd(techniques []*stratus.AttackTechnique) {
+	techniqueChan := make(chan *stratus.AttackTechnique)
+	done := make(chan bool)
+	for i := 0; i < workerCount; i++ {
+		go func() {
+			for {
+				technique, more := <-techniqueChan
+				if more {
+					stratusRunner := runner.NewRunner(technique, forceWarmup)
+					_, err := stratusRunner.WarmUp()
+					if err != nil {
+						log.Fatal(err)
```

Comment:
in terms of error handling, what happens if one of the runs returns an error while others are still running? Are they brutally stopped?

Reply:
I solved this by using a `chan(error)` so every action has a chance to send back an error. The logic is:

- We print all errors
- An error occurring for one technique doesn't prevent other techniques from being detonated/warmedup/cleanedup/reverted
- If at least one worker has an error, we exit with a non-zero status code